### DISCUSSION TITLE
Use pytest_collection_modifyitems in a plugin

### DISCUSSION THREAD

**Discussion by mdhaisne** [pytest-dev/pytest#10585] [Q&A] [answered]
Hello everyone,
I'm trying to create a pytest plugin on my own but my hook `pytest_collection_modifyitems` is neved called (at least seems like).
Below a bit of code:

foo_plugin.py
```python
class FooPlugin:
    def __init__(self, config: Config) -> None:
        self.config = config

    @hookimpl
    def pytest_collection_modifyitems(self, config: Config, items: List[Item]) -> None:
        for item in items:
            item.add_marker("foo")

    @hookimpl
    def pytest_configure(self, config: Config) -> None:
        line = f"foo(): Mark a test as a member of foo."
        config.addinivalue_line("markers", line)

```

plugin.py (entrypoint)
```python
from foo_plugin import FooPlugin

def pytest_configure(config: Config) -> None:
    fp= FooPlugin(config)
    config.pluginmanager.register(fp, "FooPlugin")
```

I've a small test using pytester and added breakpoint in both `pytest_configure` and `pytest_collection_modifyitems`. `pytest_configure` is called without problem and does it job. But my breakpoints in  `pytest_collection_modifyitems` are never triggered.
What am I missing here ? 

Thanks for your help

**Comment by The-Compiler** [comment, marked as answer]
Works for me (with your `plugin.py` as `conftest.py`). Can you try inserting a `1/0` there to make sure it's really not triggered? With a `pytest --debug` you should get a `pytestdebug.log` which could tell you more.

**Reply by mdhaisne** [reply]
Okay that was a mistake on my side ups.

I had this code
```python
def pytest_configure(config: Config) -> None:
    if config.getoption("foo"):
        fp= FooPlugin(config)
        config.pluginmanager.register(fp, "FooPlugin")
```
But I've oversimplified it because I thought that the if condition couldn't have been the cause of my issue. I should have shared the full code because the error wasn't in what I've shared.

My problem got fixed by removing the if statement. Sorry for the errors, the fault on me.

Thank you for your fast reply, appreciate it.
